### PR TITLE
feat: ctx to client API

### PR DESCRIPTION
**Pull request by Benehiko**
This is a breaking change to the Docker Client APIs as consumers of the API will be forced to add the `ctx` parameter to their callback functions.

This PR is a small piece of https://github.com/moby/moby/pull/47518. 

**- What I did**
Add ctx to the client API functions used inside the Docker CLI.

**- How I did it**

**- How to verify it**

**- Description for the changelog**

```markdown changelog
Client API callback functions `client.RequestPrivilegeFunc`, `client.ImageSearchOptions.AcceptPermissionsFunc` and `image.ImportOptions.PrivilegeFunc` now require a context parameter.
```

**- A picture of a cute animal (not mandatory but encouraged)**

### CODE REVIEW

**Review by AkihiroSuda — APPROVED** [review]

File: api/types/image/opts.go
Comment (vvoland) [inline api/types/image/opts.go:34]: Since it's a breaking change, should we first deprecate the old one, at least for one API version?

```suggestion
    // Deprecated: use [PrivilegeCtxFunc] instead
	PrivilegeFunc func() (string, error)
	
	PrivilegeCtxFunc func(context.Context) (string, error)
```

Then on the usage side, if `PrivilegeCtxFunc` is nil, call `PrivilegeFunc`.

Comment (thaJeztah) [reply api/types/image/opts.go:34]: Was looking as well at this; I know in most cases we go to the deprecate (and alias) -> remove cycle. Not sure if we always were able to do this though, and it's possible we have changed signatures for other cases (thinking of context passing etc); at least the change wouldn't go "silent".

Tricky one

**Comment by thaJeztah** [comment]
I really have a feeling something's broken in GitHub actions; I noticed this before: it shows "approval is needed" for actions to run, but it still runs most (all??) of them. It mentions "2 workflows awaiting approval", but can't see immediately... which ones didn't run?

Screenshot 2024-03-14 at 16 11 06

File: api/types/image/opts.go
Comment (thaJeztah) [reply api/types/image/opts.go:34]: Yeah, that makes sense; we're probably too far into RCs for this change.

**Review by thaJeztah — CHANGES_REQUESTED** [review]
Let me put in a "request changes" and we can merge this for v27 (after the v26 branch is cut)

**Comment by vvoland** [comment]
I think the missing ones are the `validate-pr` checks that were recently added

**Comment by Benehiko** [comment]
Would it be possible to get this one merged since v26 has been released?

**Review by vvoland — APPROVED** [review]

**Comment by Benehiko** [comment]
Could we get this merged?

**Review by laurazard — APPROVED** [review]
LGTM

**Comment by laurazard** [comment]
@thaJeztah you okay with merging this one?

**Review by thaJeztah — APPROVED** [review]
LGTM

**Pull request merged**
